orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/DeleteDeliveryConfigTask.java

```diff
@@ -0,0 +1,66 @@
+package com.netflix.spinnaker.orca.front50.tasks;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.Task;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.front50.model.DeliveryConfig;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import retrofit.client.Response;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+
+import static org.springframework.http.HttpStatus.*;
+
+@Component
+public class DeleteDeliveryConfigTask implements Task {
+
+  private Logger log = LoggerFactory.getLogger(getClass());
+
+  private Front50Service front50Service;
+  private ObjectMapper objectMapper;
+  private DeliveryConfigUtils deliveryConfigUtils;
+
+  @Autowired
+  public DeleteDeliveryConfigTask(Front50Service front50Service, ObjectMapper objectMapper) {
+    this.front50Service = front50Service;
+    this.objectMapper = objectMapper;
+    deliveryConfigUtils = new DeliveryConfigUtils(front50Service, objectMapper);
+  }
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    if (!stage.getContext().containsKey("deliveryConfigId")) {
```

Comment:
if this were `stage.mapTo()` instead with a class that has `deliverConfigId` then if we ever change the name of the field we would get a compile-time error versus a confusing runtime error. Get to harness refactor, etc functionality that tools provide

Reply:
It doesn't seem like typing the expected field brings us a ton of value - in `MonitorFront50Task` we type the context (`StageData`), but still need to check if the fields are null because if something changes in what we're sending the field doesn't get mapped. I'm happy to type this and continue with a good pattern though.

Reply:
The main thing it buys is compile-time verification and easy "find all usages", etc. I like the `stageData` @ajordens recommended